pkg/resource/deploy/step_generator.go

```diff
@@ -275,7 +275,14 @@ func (sg *stepGenerator) GenerateSteps(event RegisterResourceEvent) ([]Step, *re
 			// Determine whether the change resulted in a diff.
 			d, diffErr := sg.diff(urn, old.ID, oldInputs, oldOutputs, inputs, prov, allowUnknowns)
 			if diffErr != nil {
-				return nil, result.FromError(diffErr)
+				// If the plugin indicated that the diff is unavailable, assume that the resource will be updated and
+				// report the message contained in the error.
+				if d.Changes == plugin.DiffUnavailable {
+					d.Changes = plugin.DiffSome
+					sg.plan.ctx.Diag.Warningf(diag.RawMessage(urn, diffErr.Error()))
```

Comment:
We have the URN of the provider from the provider reference, right? can we issue the warning for that URN and not this resource? I think it would make more sense for this message to be attached to the provider since the message directly pertains to it.

Reply:
> I think it would make more sense for this message to be attached to the provider since the message directly pertains to it.

I think that this message needs to be associated with the resource: there is no visual indication that a resource and its provider are related, and it is the resource's diff that is affected.